src/wasm/WASMBuiltinOperations.h

```diff
@@ -235,19 +235,26 @@ static wasm_func_t* wasmCreateHostFunction(ExecutionState& state, Object* func,
 
 static void wasmReadImportsOfModule(ExecutionState& state, wasm_module_t* module, const Value& importObj, own wasm_extern_vec_t* imports)
 {
+    // If importObject is not undefined and not Object, throw a TypeError exception.
+    if (!importObj.isUndefined() && !importObj.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::WASM_ReadImportsError);
+    }
+
     own wasm_importtype_vec_t import_types;
     wasm_module_imports(module, &import_types);
 
-    // If module.imports is not empty, and importObject is undefined, throw a TypeError exception.
-    if (import_types.size > 0 && importObj.isUndefined()) {
+    // If module.imports is not empty, and importObject is not Object, throw a TypeError exception.
+    if (import_types.size > 0 && !importObj.isObject()) {
         wasm_importtype_vec_delete(&import_types);
-        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().WebAssembly.string(), false, state.context()->staticStrings().Module.string(), ErrorObject::Messages::WASM_ReadImportsError);
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, ErrorObject::Messages::WASM_ReadImportsError);
     }
 
     if (!importObj.isObject()) {
         wasm_importtype_vec_delete(&import_types);
         return;
```

Comment:
There are calling ```dalete[] imports->data``` In the other return statement of this function. Isn't it necessary here as well?